experimental/dds/tree2/src/core/tree/delta.ts

```diff
@@ -167,226 +95,95 @@ export type ProtoNode = ITreeCursorSynchronous;
 	 */
-	readonly detachId?: DetachedNodeId;
+	readonly attach?: DetachedNodeId;
 }
 
 /**
- * When set, indicates that the inserted content is replacing some existing content.
+ * Represents a list of changes to some range of nodes. The index of each mark within the range of nodes, before
+ * applying any of the changes, is not represented explicitly.
+ * It corresponds to the sum of `mark.count` values for all previous marks.
  * @alpha
  */
-export interface OldContent<TTree = ProtoNode> {
-	/**
-	 * Modifications to the old content.
-	 */
-	readonly fields?: FieldMarks<TTree>;
-	/**
-	 * The ID to assign the first node being replaced.
-	 * Subsequent replaced nodes should be assigned incrementing IDs.
-	 */
-	readonly detachId: DetachedNodeId;
-}
+export type MarkList<TTree = ProtoNode> = readonly Mark<TTree>[];
 
 /**
- * Describes the insertion of a contiguous range of node.
+ * A globally unique ID for a node in a detached field.
  * @alpha
  */
-export interface Insert<TTree = ProtoNode>
-	extends HasModifications<TTree>,
-		CanReplaceContent<TTree> {
-	readonly type: typeof MarkType.Insert;
-
-	// TODO: use a single cursor with multiple nodes instead of array of cursors.
-	/**
-	 * Must be of length 1 when `fields` is populated.
-	 */
-	readonly content: readonly TTree[];
-
-	/**
-	 * ID associated with the tree being built.
-	 * If not specified, a random unique ID will be used.
-	 */
-	readonly buildId?: DetachedNodeId;
-
-	/**
-	 * The ID to assign the first node being replaced.
-	 * Subsequent replaced nodes should be assigned incrementing IDs.
-	 * Populated iff the insertion is transient.
-	 */
-	readonly detachId?: DetachedNodeId;
+export interface DetachedNodeId {
+	readonly major?: string | number;
+	readonly minor: number;
 }
 
 /**
- * Describes the restoration of a contiguous range of node.
  * @alpha
  */
-export interface Restore<TTree = ProtoNode> extends CanReplaceContent<TTree> {
-	readonly type: typeof MarkType.Restore;
-	/**
-	 * Must be 1 when `fields` is populated.
-	 */
-	readonly count: number;
-
-	readonly newContent: {
-		/**
-		 * The ID assigned to the first node being restored.
-		 * Subsequent nodes should be assigned incrementing IDs.
-		 */
-		readonly restoreId: DetachedNodeId;
-
-		/**
-		 * The ID to assign the first node being replaced.
-		 * Subsequent replaced nodes should be assigned incrementing IDs.
-		 * Populated iff the restoration is transient.
-		 */
-		readonly detachId?: DetachedNodeId;
-	};
-
-	/**
-	 * Modifications to the new content.
-	 */
-	readonly fields?: FieldMarks<TTree>;
-}
+export type FieldMap<T> = ReadonlyMap<FieldKey, T>;
 
 /**
- * Uniquely identifies a node in the scope of a delta.
- * Often accompanied with a count that indicates a set of nodes whose IDs are contiguous.
  * @alpha
  */
-export interface MoveId extends Opaque<Brand<number, "delta.MoveId">> {}
+export type FieldsChanges<TTree = ProtoNode> = FieldMap<FieldChanges<TTree>>;
```

Comment:
This name is awkward. Could we just call this `FieldMap`? Can the existing `FieldMap` alias just be inlined?